src/actix/api_key.rs

```diff
@@ -70,7 +73,16 @@ where
 
         if let Some(key) = req.headers().get("api-key") {
             if let Ok(key) = key.to_str() {
-                if constant_time_eq(self.api_key.as_bytes(), key.as_bytes()) {
+                let is_allowed = if let Some(ref auth_keys) = self.auth_keys {
+                    auth_keys.can_write(key) || is_read_only(&req) && auth_keys.can_read(key)
```

Comment:
I prefer explicit brackets here because this is security sensitive:

```suggestion
                    auth_keys.can_write(key) || (is_read_only(&req) && auth_keys.can_read(key))
```